packages/cf.js/API_REFERENCE.md

```diff
@@ -0,0 +1,54 @@
+# API V0.0.2
+## `cf.js`
+
+- `Client`
+    - Properties
+        - `nodeProvider`
+    - Instance methods
+        - `getAppInfos(): AppInfo[]`
+        - `createAppFactory(appDefinition: AppDefinition): AppFactory`
+    - Client lifecycle
+        - `on(eventType, callback)`
+            - EventType: proposeInstall, install, rejectInstall
```

Comment:
Are these all event types?

Reply:
Yeah

Reply:
Should we also have an `error` event type? For instance, if the user submits an invalid `proposeInstall`.

Reply:
I would say so @patience-tema-baron

Reply:
Who would be a recipient of such an event?

Ideally clients (by which I mean the node) should do verification to confirm the validity of a proposal on sending/receiving.

Let's say the receiver of a proposal is notified via the `proposeInstall` event, and upon checking the proposal realizes the proposal is invalid. That seems to just be a `rejectInstall` response, with specific details about the rejection in the payload being passed back.

On the other hand, let's say the sender wants to send a proposal (which is invalid). Ideally the sender is immediately notified (i.e. awaits the "send-proposal" call) via a verification process by the node that the proposal is invalid.

Reply:
Exactly. I'd imagine the `error` notification to be this latter case, useful for disambiguating a system response from a peer response. That being said, perhaps they could both fall under the umbrella of `rejectInstall` and (as you suggest) we would just differentiate between them with the payload.